example/screens/ConnectWithRoomManagerScreen.tsx

```diff
@@ -0,0 +1,152 @@
+import React, {useState} from 'react';
+import {
+  Dimensions,
+  Image,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import {connectScreenLabels} from '../types/ComponentLabels';
+import {useJellyfishClient} from '@jellyfish-dev/react-native-client-sdk';
+import {Button, TextInput, DismissKeyboard} from '../components';
+import {NativeStackScreenProps} from '@react-navigation/native-stack';
+import {AppRootStackParamList} from '../navigators/AppNavigator';
+import type {BottomTabScreenProps} from '@react-navigation/bottom-tabs';
+import {TabParamList} from '../navigators/AppNavigator';
+import {CompositeScreenProps} from '@react-navigation/native';
+import {usePermissionCheck} from '../hooks/usePermissionCheck';
+
+type Props = CompositeScreenProps<
+  BottomTabScreenProps<TabParamList, 'ConnectWithRoomManager'>,
+  NativeStackScreenProps<AppRootStackParamList>
+>;
+
+const {URL_INPUT, TOKEN_INPUT, CONNECT_BUTTON} = connectScreenLabels;
+
+async function getJellyFishServer(
+  roomManagerUrl: string,
+  roomName: string,
+  userName: string,
+) {
+  const url = roomManagerUrl.endsWith('/')
+    ? roomManagerUrl
+    : roomManagerUrl + '/';
+  const result = await fetch(
+    `${url}rooms/${roomName.trim()}/users/${userName.trim()}`,
+  );
+
+  const tokenData = (await result.json()) as {
+    url: string;
+    token: string;
+  };
+
+  return {
+    jellyfishUrl: tokenData.url,
+    token: tokenData.token,
+  };
+}
+
+const ConnectScreen = ({navigation}: Props) => {
+  const {connect} = useJellyfishClient();
+  const [connectionError, setConnectionError] = useState<string | undefined>(
+    undefined,
+  );
+  const [loading, setLoading] = useState(false);
+
+  const [roomManagerUrl, setRoomManagerUrl] = useState(
+    process.env.ROOM_MANAGER_URL ?? '',
+  );
+  const [roomName, setRoomName] = useState('');
+  const [userName, setUserName] = useState('');
+
+  usePermissionCheck();
+
+  const onTapConnectButton = async () => {
+    try {
+      setConnectionError(undefined);
+      setLoading(true);
+      const {jellyfishUrl, token} = await getJellyFishServer(
+        roomManagerUrl,
+        roomName,
+        userName,
+      );
+
+      await connect(jellyfishUrl, token);
+
+      navigation.navigate('Preview');
+    } catch (e) {
+      const message =
+        'message' in (e as Error) ? (e as Error).message : 'Unknown error';
+      setConnectionError(message);
+    } finally {
+      setLoading(false);
```

Comment:
When testing on Android, I've encountered a bug where if connecting to the room manager without `/api` appended to its URL, the client gets 404, no error message shows up and the button never becomes active again.
<img width="949" alt="Screenshot 2024-04-18 at 16 20 07" src="https://github.com/jellyfish-dev/react-native-client-sdk/assets/31112335/c3af56e7-ade6-4b92-8c24-a4f961138481">

Reply:
Indeed, there was a bug related with new version of Room Manger